app/models/user.rb

```diff
@@ -38,4 +38,13 @@ class User < ApplicationRecord
   def full_name
     "#{first_name} #{last_name}"
   end
+
+  def update_with_password(params = {})
+    if params[:password].blank?
+      params.delete(:password)
+      params.delete(:password_confirmation) if params[
```

Comment:
Why do you check if the `password_confirmation` is blank? I guess that, if the password is empty, we should also drop the `password_confirmation`, automatically.
If not, I think this could lead to the case when only the `password_confirmation` is set.